core/src/executor/alter/table.rs

```diff
@@ -146,8 +219,84 @@ pub async fn drop_table<T: GStore + GStoreMut>(
             schema.ok_or_else(|| AlterError::TableNotFound(table_name.to_owned()))?;
         }
 
+        let schemas = storage.fetch_all_schemas().await?;
+        let referring_children: Vec<ReferringChild> = schemas
+            .into_iter()
+            .flat_map(
+                |Schema {
+                     table_name: referring_table_name,
+                     foreign_keys,
+                     ..
+                 }| {
+                    foreign_keys
+                        .into_iter()
+                        .filter_map(
+                            |ForeignKey {
+                                 name: constraint_name,
+                                 referred_table,
+                                 ..
+                             }| {
+                                if &referred_table == table_name
+                                    && &referring_table_name != table_name
+                                {
+                                    return Some(ReferringChild {
+                                        table_name: referring_table_name.clone(),
+                                        constraint_name,
+                                    });
+                                }
+
+                                None
+                            },
+                        )
+                        .collect::<Vec<_>>()
+                },
+            )
+            .collect();
+
+        match (!referring_children.is_empty(), cascade) {
+            (true, false) => {
+                return Err(AlterError::CannotDropTableParentOnReferringChildren {
+                    parent: table_name.into(),
+                    referring_children,
+                }
+                .into());
+            }
+            (true, true) => {
+                for ReferringChild {
+                    constraint_name, ..
+                } in referring_children
+                {
+                    let mut schema = storage
+                        .fetch_schema(table_name)
+                        .await?
+                        .ok_or_else(|| AlterError::TableNotFound(table_name.to_owned()))?;
+                    schema
+                        .foreign_keys
+                        .retain(|foreign_key| foreign_key.name != constraint_name);
+                    storage.insert_schema(&schema).await?;
+                }
+            }
+            (false, _) => {}
+        }
```

Comment:
cascade check is only required at the first, so.. simple if statement would be enough.
```rust
 if !referring_children.is_empty() && !cascade {
    return Err(AlterError::CannotDropTableParentOnReferringChildren {
        parent: table_name.into(),
        referring_children,
    }
    .into());
}

for ReferringChild {
    constraint_name, ..
} in referring_children
{
    let mut schema = storage
        .fetch_schema(table_name)
        .await?
        .ok_or_else(|| AlterError::TableNotFound(table_name.to_owned()))?;
    schema
        .foreign_keys
        .retain(|foreign_key| foreign_key.name != constraint_name);
    storage.insert_schema(&schema).await?;
}
```